src/providers/db2/qgsdb2dataitems.cpp

```diff
@@ -0,0 +1,467 @@
+      delete item;
+      continue;
+    }
+    addChildItem( item, true );
+  }
+}
+
+QVector<QgsDataItem*> QgsDb2ConnectionItem::createChildren()
+{
+  QVector<QgsDataItem*> children;
+
+  QString connInfo;
+  QString errorMsg;
+  bool success = QgsDb2ConnectionItem::ConnInfoFromSettings( mName, connInfo, errorMsg );
+  if ( !success )
+  {
+    QgsDebugMsg( "settings error: " + errorMsg );
+    children.append( new QgsErrorItem( this, errorMsg, mPath + "/error" ) );
+    return children;
+  }
+
+  mConnInfo = connInfo;
+  QgsDebugMsg( "mConnInfo: '" + mConnInfo + "'" );
+
+  QSqlDatabase db = QgsDb2Provider::GetDatabase( connInfo, errorMsg );
+  if ( errorMsg.isEmpty() )
+  {
+    QString connectionName = db.connectionName();
+    //children.append( new QgsFavouritesItem(this, "connection successful", mPath + "/success"));
+    QgsDebugMsg( "DB open successful for connection " + connectionName );
+  }
+  else
+  {
+    children.append( new QgsErrorItem( this, errorMsg, mPath + "/error" ) );
+    QgsDebugMsg( "DB not open " + errorMsg );
+    return children;
+  }
+  QString connectionName = db.connectionName();
+  QgsDb2GeometryColumns db2GC = QgsDb2GeometryColumns( db );
+  int sqlcode = db2GC.open();
+
+  /* Enabling the DB2 Spatial Extender creates the DB2GSE schema and tables,
+     so the Extender is either not enabled or set up if SQLCODE -204 is returned. */
+  if ( sqlcode == -204 )
+  {
+    children.append( new QgsErrorItem( this, tr( "DB2 Spatial Extender is not enabled or set up." ), mPath + "/error" ) );
+    return children;
+  }
+  else if ( sqlcode != 0 )
+  {
+    children.append( new QgsErrorItem( this, db.lastError().text(), mPath + "/error" ) );
+    return children;
+  }
+  QgsDb2LayerProperty layer;
+
+  //QVector<QgsDataItem*> newLayers;
+  while ( db2GC.populateLayerProperty( layer ) )
+  {
+    QgsDb2SchemaItem* schemaItem = NULL;
+    Q_FOREACH ( QgsDataItem *child, children )
+    {
+      if ( child->name() == layer.schemaName )
+      {
+        schemaItem = ( QgsDb2SchemaItem* )child;
+        break;
+      }
+    }
+
+    if ( !schemaItem )
+    {
+      schemaItem = new QgsDb2SchemaItem( this, layer.schemaName, mPath + "/" + layer.schemaName );
+      QgsDebugMsg( "Adding Schema Item : " + layer.schemaName + " " + mPath + "/" + layer.schemaName + " type=" + layer.type
+                   + " srid=" + layer.srid + " table=" + layer.tableName + "(" + layer.geometryColName + ")"
+                 );
+      children.append( schemaItem );
+    }
+
+    QgsDb2LayerItem* added = schemaItem->addLayer( layer, true );
+
+    if ( added )
+    {
+      QgsDebugMsg( " DB2 adding layer to schema item: " + added->name() );
+      //children.append(added);
+    }
+    else
+    {
+      QgsDebugMsg( " DB2 layer not added " );
+    }
+  }
+  return children;
+}
+
+QList<QAction*> QgsDb2ConnectionItem::actions()
+{
+  QList<QAction*> lst;
+
+  QAction* actionRefresh = new QAction( tr( "Refresh" ), this );
```

Comment:
Please rename these actions to "Refresh Connection", "Edit Connection...", "Delete Connection", to make it clear that deleting etc is deleting JUST the connection, not the database itself.